Library/Homebrew/test.rb

```diff
@@ -17,6 +17,15 @@
   ENV.extend(Stdenv)
   ENV.setup_build_environment
 
+  # This ensures all executables in test do are called with a full path, i.e.
+  # bin/"wget", or Formula["openssl"].opt_bin/"openssl" rather than just "wget".
+  # It's very likely to break tests, so phase it in via CI rather than user
```

Comment:
`test` runs under `stdenv`, with the idea being that we test what the user has installed rather than our pristine happy environment. Also means there's no automatic shunting of `keg_only` dependencies into the `$PATH`. So for example, with `libressl`:

```
==> openssl dgst -sha256 -out checksum.txt testfile.txt
```
All we'd be doing there is testing the system OpenSSL works, which is nice, but not really what we want.

The start of my PATH as setup by my shell profile is:
```
/Users/xyz/perl5/bin:/usr/local/opt/openssl/bin:/usr/local/opt/curl/bin:/Users/xyz/.opam/system/bin:/usr/local/var/rbenv/shims:/usr/local/bin:/usr/local/sbin
```
Which means if I build a piece of software locally that happens to reside in anything prior to `/usr/local/bin` or `/usr/local/sbin` I'm not really testing the software intended. This can get particularly fun if you're testing stuff on 10.8 where `/usr/local/bin` wasn't yet dominant in the `$PATH`.